### DISCUSSION TITLE
Ranimated 2 and Debugging

### DISCUSSION THREAD

**Discussion by shawnFallon** [software-mansion/react-native-reanimated#3669] [Q&A]
I am having a “discussion" with another developer as he would like to remove reanimated 2 for debugging purposes.  He has said that no company (like Shopify) would use reanimated 2 at the expense of their debugger.  I said we can use flipper and carry on.

I was wondering if anyone else would like to shine some light on this subject for me. My understanding is that there are many many well know apps that use reanimated 2.  And I certainly don't want to remove my super sick animations.

Is there a way to use reanimated 2 and a "source level debugger"....

Are there other companies out there that are using reanimated 2?  

If anyone can chime in on this I can build up my case :)

Cheers

**Comment by tomekzaw** [comment]
Hey, Reanimated 2 works well with all kinds of debuggers except for Chrome Debugger (remote) which executes the JS bundle of RN app on your computer instead of on the device (which by the way doesn't work with any JSI-enabled library and currently there's plenty of those).

The recommended way to debug RN apps in 2022 is to use Hermes debugger via Flipper or Google DevTools. Reanimated not only doesn't break debugging the main JS context but also exposes a Flipper plugin to debug the secondary worklet UI runtime.

Make sure to check out this very insightful section in our docs on debugging React Native apps with Reanimated: https://docs.swmansion.com/react-native-reanimated/docs/next/guide/debugging/
